pkg/yttlibrary/json.go

```diff
@@ -49,26 +51,33 @@ func (b jsonModule) Encode(thread *starlark.Thread, f *starlark.Builtin, args st
 	val = orderedmap.Conversion{yamlmeta.NewGoFromAST(val)}.AsUnorderedStringMaps()
 
 	var valBs []byte
+	buffer := bytes.NewBuffer(valBs)
 	indent, err := core.Int64Arg(kwargs, "indent")
 	if err != nil {
 		return starlark.None, err
 	}
-
 	if indent < 0 || indent > 8 {
 		// mitigate https://cwe.mitre.org/data/definitions/409.html
 		return starlark.None, fmt.Errorf("indent value must be between 0 and 8")
 	}
 
+	escapeHTML, err := core.BoolArg(kwargs, "escape_html", true)
+	if err != nil {
+		return starlark.None, err
+	}
+
+	encoder := json.NewEncoder(buffer)
 	if indent > 0 {
-		valBs, err = json.MarshalIndent(val, "", strings.Repeat(" ", int(indent)))
-	} else {
-		valBs, err = json.Marshal(val)
+		encoder.SetIndent("", strings.Repeat(" ", int(indent)))
 	}
-	if err != nil {
+	encoder.SetEscapeHTML(escapeHTML)
+
+	if err := encoder.Encode(val); err != nil {
 		return starlark.None, err
 	}
 
-	return starlark.String(string(valBs)), nil
+	res := strings.TrimSuffix(buffer.String(), "\n")
```

Comment:
why is the trimsuffix needed now?

Reply:
Otherwise the json encoder always add an extra `\n` at the end of the encoded json.

I use `TrimSuffix` for backward compatibility. (and make tests happy)